docs/architecture/adr-0025-user-access-scopes.md

```diff
@@ -142,6 +142,51 @@ Usage at the endpoint level might look like:
 - requires a bunch of columns to be added (most likely to the `User` data-model/table)
 - will require some machinery to map `ObjectType`s to columns
 
+### The More Granular Variation
+
+This solution is similar to _The Civilized Solution_ in that it would use an absence of scopes for a certain domain to imply full access. For example, if a user has no `project` scopes, then they are granted full access to project operations. This solution differs from the others in how the scopes are treated at the endpoint level. For example, if that user instead has a scope of `project:read`, they would be allowed to interact with project read end-points. This is in contrast to the other approaches that invert the process and determine what operations are _not_ allowed using scopes.
+
+- could be implemented with a single scopes column or a column for each domain (project, scenes, etc.), where the columns would contain a list of applied scopes for that user
+- Scope types would be defined within a `ScopeType` to enforce values
+- Each API endpoint that is affected would do a lookup on the requesting user and ensure that if a scope is present for the relevant domain that the necessary scope was present
+
+Usage at the endpoint level might look like:
+
+```scala
+  def createProject: Route = authenticate { user =>
+    authorizeAsync {
+      UserDao
+        .hasScope(ObjectType.Project, ScopeType.Read)
+        .transact(xa)
+        .unsafeToFuture
+    } {
+      entity(as[Project.Create]) { newProject =>
+        onSuccess(
+          ProjectDao
+            .insertProject(newProject, user)
+            .transact(xa)
+            .unsafeToFuture) { project =>
+          complete(StatusCodes.Created, project)
+        }
+      }
+    }
+  }
+```
+
+#### Advantages
+
+- easy to add new scopes as needed
+- promotes consistency by narrowing `ScopeType` members
+- encourages a more granular approach to scoping versus larger all-encompassing scopes types such as `readOnly`
+- much more intuitive at the endpoint level
+
+#### Disadvantages
+
+- multiple scopes per domain open the possibility for conflicts but those could be avoided with sufficiently granular scopes types
```

Comment:
I still don't really know where conflicts are going to come from here. I think we can ask, when creating a scope, "does this do two things," and if the answer is yes, split into two scopes, and avoid conflicts forever. Or if we're extremely careful we can probably come up with non-conflicting scopes similar to what GitHub does https://developer.github.com/apps/building-oauth-apps/understanding-scopes-for-oauth-apps/#available-scopes

I don't know that it's worth removing mention of conflicts, but I still think they're really unlikely

Reply:
Thinking about this a bit more, if the scopes being added are all permissive (grant permissions) then conflicts aren't really possible. The concept of a conflict can only really occur when you mix permissive and restrictive scopes, which we can just not do. I'm going to remove mention of conflicts on this option.